client/state/reader/watermarks/watermark-schema.js

```diff
@@ -0,0 +1,10 @@
+/** @format */
+
+export default {
+	type: 'object',
+	additionalProperties: false,
+	patternProperties: {
+		// todo: does this match everything? thats what i want
```

Comment:
Can we remove this todo?

Reply:
i'm happy i added the todo, i still need to verify it

Reply:
i think you can just leave properties and patternProperties off entirely and set `additonalProperties: { type: 'number' }`?

Reply:
either way, we should have tests for the schema.

Reply:
adding tests for deserialization now. that will test the schema